src/pages/scientificServices/pipelines/utils/upload-utils.test.ts

```diff
@@ -127,4 +127,51 @@ describe('upload-utils', () => {
       expect(mockXHR.send).toHaveBeenCalledWith(inputFile.slice(5));
     });
   });
+
+  describe('initiateResumableUpload', () => {
+    it('initiates a resumable upload session', async () => {
+      (global.fetch as jest.Mock).mockResolvedValueOnce({
+        status: 200,
+        ok: true,
+        headers: {
+          get: () => 'http://signed.url/session',
+        },
+      });
+
+      const mockSetUploadState = jest.fn();
+      const inputName = 'testInput';
+      const inputFile = new File(['abcdefghij'], 'foo.txt');
+      const signedUrl = 'http://signed.url/upload';
+
+      const mockXHR = new MockXMLHttpRequest();
+      mockXHR.addEventListener = jest.fn((event, callback) => {
+        if (event === 'load') {
+          setTimeout(() => callback({ status: 200 }), 0);
+        }
+      });
+
+      Object.defineProperty(mockXHR, 'status', {
+        value: 200,
+        writable: true,
+      });
+
+      global.XMLHttpRequest = jest.fn(() => mockXHR) as any;
+
+      await initiateResumableUpload(inputName, inputFile, signedUrl, mockSetUploadState);
+
+      expect(global.fetch).toHaveBeenCalledWith(
+        signedUrl,
+        expect.objectContaining({
+          method: 'POST',
+          headers: { 'x-goog-resumable': 'start', 'Content-Type': 'application/octet-stream' },
+        })
+      );
+
+      expect(global.XMLHttpRequest).toHaveBeenCalled();
+      expect(mockXHR.open).toHaveBeenCalledWith('PUT', 'http://signed.url/session');
+      expect(mockXHR.setRequestHeader).toHaveBeenCalledWith('Content-Type', 'application/octet-stream');
+      expect(mockXHR.setRequestHeader).toHaveBeenCalledWith('Content-Range', 'bytes 0-9/10');
```

Comment:
whats this byte range thing?

Reply:
for our use-case and in this example test scenario, we're basically just telling google that we're intending to upload bytes 0-9, out of a total of 10 bytes.

where this really comes into play is if an upload is resumed after failure, then we get more specific with the byte range. but for the initial upload attempt, we try the full object as a single chunk.